### DISCUSSION TITLE
Libre2 Not Connected

### DISCUSSION THREAD

**Discussion by Brandters** [NightscoutFoundation/xDrip#3555] [Q&A]
Suddenly my Libre2 just stopped working, it's unable to connect. I've done everything I can think of, first I tried to stop and start the sensor, and I can get readings, OOP2 is working as it should, but it refuses to reconnect after the first scan. I can scan it with NFC and get readings. After that, I removed all my other BT devices just to see if anything was messing with the connection it didn't do anything. After that, I restarted the collector but didn't do anything. I forgot the device and resynced it, didn't work I even went as far as to remove Xdrip and reinstall it so I had a clean slate, doesn't change anything. If I use the official app the connection works and I get readings.
I restarted the phone and the Bluetooth, wifi and more but nothing helps.

It just stopped working. 

Any ideas?
It's a Samsung S22, I tested the Samsung fix but that didn't fix it.

Logs even state that Bluetooth connection is enabled. 

![Screenshot_20240701_214827_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/35875020/aa363824-2b8a-4dfe-9e87-df54cf3a22a1)
![Screenshot_20240701_214838_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/35875020/694301da-8b10-4ad7-a295-580d19752c33)
![Screenshot_20240701_215124_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/35875020/645d5ce7-5a64-4ac4-82f2-2da358240e3d)

**Comment by parapenT1sta** [comment]
Check if Battery Optimization is not enabled.

I only can make it working on my  Z Fold 4  phone with these settings: (forget device and retry to pair it again)

![Screenshot_20240702_073542_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/608650/f8f4e990-96dc-4eec-859e-585dd851d36b)

**Reply by Brandters** [reply]
Yes, batter optimization is disabled, it still drops connection from time to time, it will connect after 1h-2h or so. Since I had this issue I thought it had to do with the sensors itself, I've swapped it out and it work perfectly fine until day 2 and at that point it started dropping connections again. I've repaired the device many times as of today but it never stays connected like it did before. Also, I've noted the drops happen more frequently when the phone jumps between mobile and wifi data. So when I get home it stops working for hours some times. And some times it works as it should. Very annoying and it started with the latest sensors I've received.

**Comment by PhilTU** [comment]
does OOP2 screen show received values ? what else is connected to the Bluetooth ?

**Reply by Brandters** [reply]
Yes, all values work, it's "just" the sensor that drops connection from time to time.
I've a Samsung Watch 5, nothing else at the time of the issues, when I've my BT headphones connected I've not noticed it is dropping connection as often, but I think that is just random luck to be fair.
NFC works everytime, but even after a scan it most of the time does not stay connected, when it drops connection it can take hours until it manages to reconnect.
